sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoinExec.scala

```diff
@@ -60,6 +60,26 @@ case class BroadcastHashJoinExec(
     }
   }
 
+  override def outputPartitioning: Partitioning = {
+    def buildKeys: Seq[Expression] = buildSide match {
+      case BuildLeft => leftKeys
+      case BuildRight => rightKeys
+    }
+
+    joinType match {
+      case _: InnerLike =>
+        streamedPlan.outputPartitioning match {
+          case h: HashPartitioning =>
+            PartitioningCollection(Seq(h, HashPartitioning(buildKeys, h.numPartitions)))
```

Comment:
Are you sure this is right??
Say "t1 join t2 on t1.a = t2.b" and t1 is partitioned on an unrelated column "c", then we can say the join result is partitioned on "b" as well???

Reply:
This is only true when t1 is partitioned on "a" then because a == b, so "t1 join t2" is partitioned on "b".

Reply:
Yes, you are right. The streamed join keys should match with the streamed output partitioning for this to be true.

Reply:
I added this check, but when `BroadcastHashJoinExec` executes, `t1`'s streamed side should have been shuffled by `t1.a` even if `t1` was originally partitioned by an unrelated column `c`, no?